cmd/kube-apiserver/app/testing/testserver.go

```diff
@@ -177,7 +180,12 @@ func StartTestServer(t Logger, instanceOptions *TestServerInstanceOptions, custo
 
 	featureGate := utilfeature.DefaultMutableFeatureGate
 	featureGate.DeferErrorsToValidation(true)
-	effectiveVersion := utilversion.DefaultEffectiveVersionRegistry.EffectiveVersionForOrRegister(utilversion.ComponentGenericAPIServer, utilversion.K8sDefaultEffectiveVersion())
+	binaryVersion := "1.31"
```

Comment:
Would this need to be updated with each release? Can we avoid that?

Reply:
It does not have to be updated for each release. If a new api is introduced later, in that particular test, they would need to change this version. This version is set to avoid the default `0.0`, there is no way around it.

Reply:
> If a new api is introduced later, in that particular test, they would need to change this version.

My concern is that this will result in this needing to be updated regularly, and may become a headache for contributors who are unaware of this field and need to spend time tracking down why a feature is disabled in the test server.

Could we instead use [`version.Get()`](https://github.com/kubernetes/kubernetes/blob/69e332560d0357b5b4db7d5461b12b5ce0be6c3f/staging/src/k8s.io/component-base/version/version.go#L28) to set this to the current version most of the time?

Reply:
24 integration tests would fail if I use [version.Get()](https://github.com/kubernetes/kubernetes/blob/69e332560d0357b5b4db7d5461b12b5ce0be6c3f/staging/src/k8s.io/component-base/version/version.go#L28), due to various APIs with introduced annotations > 0.0. I would expect this to be worse with Alex's change.

Reply:
Sorry, just trying to understand here. The problem that Get returns 1.29 (or 1.30) right now and we need 1.31 to enable everything introduced?  Could this default to `version.Get() + 2 minor versions` or something?

Reply:
'version.Get()' returns 0.0 in test. We need something more sensible

Reply:
Ok, that explains it. I think I can live with a constant for now. I'd like to track this one for follow up..